dask_planner/src/sql/optimizer/join_reorder.rs

```diff
@@ -0,0 +1,392 @@
+//! Join reordering based on the paper "Improving Join Reordering for Large Scale Distributed Computing"
+//! https://ieeexplore.ieee.org/document/9378281
```

Comment:
Do we have a link to the modified join reorder algorithm implemented in spark-rapids that we could also include here?